src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.cs

```diff
@@ -343,6 +343,10 @@ await handler.InvokeAsync(nameof(IHybridWebView.EvaluateJavaScriptAsync),
 			{
 				invokeJavaScriptRequest.SetResult(null);
 			}
+			else if (invokeJavaScriptRequest.ReturnTypeJsonTypeInfo.Type == typeof(string))
+			{
+				invokeJavaScriptRequest.SetResult(stringResult);
+			}
```

Comment:
@Eilon do you think this will have issues when just comparing Type?

Reply:
Hmm perhaps? But how about other 'primitive' types - are there others that don't work and/or need code changes? Like ints, bool, etc.?

Reply:
The existing tests seem to pass:

* `InvokeJavaScriptMethodWithParametersAndResult` returns a number (decimal)
* `InvokeJavaScriptMethodWithParametersAndComplexResult` returns a complex object

I think the issue is that when you are deserializing a string, it expects it to be quoted - so this would mean you returned a string with quotes (`"\"value\""`) in JS which is probably not what you want?

Reply:
So far, it looks like only string needed a special case - probably because the communication mechanism is a string already. And it is probably because of the code in `__TriggerAsyncCallback`:

```js
        // Make sure the result is a string
        if (result && typeof (result) !== 'string') {
            result = JSON.stringify(result);
        }
```

We are special casing strings in JS, so we have to mirror it in C#. We probably can "fix" this another way by removing this JS - but that would be breaking for those folks who update MAUI but keep the same old js file.

Reply:
Ah then I think that's fine. Ultimately I think if the scenarios that customer care about work, and the code looks legit, then we're good.

Reply:
I found some issues with all these special cases, if I remove them then it all works in all [reasonable testing matrix] cases.